### ISSUE TITLE
Design Overhall

### ISSUE THREAD

**Issue by gnunicorn** [rust-lang/arewewebyet#1]
I want to give the website a complete makeover. Among other things I'd like to:
- [x] **adapt design from rust-lang.org**: colors, fonts, etc – to give it more of the same-family-style
- [x] **drop table layout**: the content is very inaccessible right now
- [x] **restructure content**: I am thinking of restructuring the server/client-differentiation into more specific parts, maybe adding a table of contents
- [x] **redo into markdown**: the table-layout in html is ... not really nice to contribute to, I'd rather have this in simplified markdown
- [x] [add comments/reviews per packaged](https://github.com/bashyHQ/arewewebyet/issues/4)
- [x] small indicator giving insight into the specific topic.
- [x] [add news](https://github.com/bashyHQ/arewewebyet/issues/3)
- [x] feature curators
- [x] explain features of the website 
- [ ] proper review
- [ ] relaunch news bulletin
- [ ] add curators

**Comment by gnunicorn** [comment]
First updates can be see in the [rewrite-branch](https://github.com/bashyHQ/arewewebyet/tree/rewrite):

![screen shot 2016-02-10 at 01 10 44](https://cloud.githubusercontent.com/assets/40496/12925897/32589220-cf93-11e5-8b75-9824c8c2fbae.png)

As visible I streamlined the design a bit and embed some cargo info per package rather than making our own (rather general) guesses. I also took the time to write a little script which fetches some crates info for each package and uses that to directly link to the homepage and show the proper info.

---

I further more decided to split things up into topic, so we can have longer lists with more info for each and can give a more in-depth look into the matter. For example for databases:

![screen shot 2016-02-10 at 01 13 53](https://cloud.githubusercontent.com/assets/40496/12925966/98b1e7d8-cf93-11e5-8cfb-cb14c5648bd9.png)

**Comment by gnunicorn** [comment]
![screen shot 2016-02-10 at 22 29 02](https://cloud.githubusercontent.com/assets/40496/12951639/d8cfd386-d045-11e5-8f27-d3b539ad83bc.png)

Update:
- improved package-info block; neater, better design
- added Topic-Maturity-Indicator
- added automatic all-topic-list on bottom of main-page
- added more topics/sections; Crypto, i18n, JSON, encoding, compression, email, soap, syndication, utils, templating, logging, oauth, services.

**Comment by gnunicorn** [comment]
I just pushed the latest [rewrite version into another repo](https://github.com/bashyHQ/arewewebyet-rewrite) to be able host it under a different subdomain. The preview should be available soon under rewrite.arewewebyet.org 

//cc @Manishearth

**Comment by Manishearth** [comment]
I'm still unsure about "though the ecosystem isn’t mature yet" and "Getting there". Those comments are from more than a year ago, and a _lot_ has changed in that time.

This is important because people keep linking to arewewebyet, seeing the "getting there", and determining that Rust isn't ready; even though it mostly is. 

@steveklabnik Do you have a better way of describing how "web" we are? The only major thing missing is async IO in hyper, which should happen soon. And then there are a bunch of minor things.

**Comment by Manishearth** [comment]
But the design is nice!

**Comment by gnunicorn** [comment]
> I'm still unsure about "though the ecosystem isn’t mature yet" and "Getting there". Those comments are from more than a year ago, and a lot has changed in that time.

I am totally happy to change this text. 

---

To be honest though, although stuff has improved, my view on the matter (since starting this whole rewrite) has rather diminished. Looking at what is missing in the ecosystem compared to what you have in Ruby, Python or Node (I am doing most of my work with right now), I'd not advice people to use rust for proper web-development yet. Backend services and APIs, maybe, but frontend/end user facing stuff – nope, it is just too much extra work (just looking at how i18n is barely existent). 

I guess it depends on what you define as "web" ;) .

> But the design is nice!

Thanks!

**Comment by steveklabnik** [comment]
Your application tier as "just a backend service" is the new/hot/standard way of building web services depending on who you ask, though, so it might be worth delving into specifics here.

**Comment by gnunicorn** [comment]
@steveklabnik I agree. But in that is the case, for that Database/ORM and moreover external-services are still very much lacking.

So if your system is a specific microservice backend/API that doesn't require many other services, only responds to other computers (via XML/JSON) rather than to humans and you are happy to write SQL directly, then yes – we are web. You can definitely do stuff with it and it is mature for those cases. But for some cases (like replace your Rails/Flask/Django) I'd give it some more time...

that's why I think a more nuanced first paragraph would be more helpful than a "we are approaching".  What do you think?

**Comment by steveklabnik** [comment]
Maybe I've drank the Diesel kool-aid too much :wink: 

Agree on external services though. Nuance is good.

**Comment by gnunicorn** [comment]
I pushed a new version – it is more text now. I've put it a bit more positive than the tone I just used here before. But I'd really appreciate a thorough review – I am sure we can make this more brief.

**Comment by Manishearth** [comment]
Ah, I love this!

**Comment by steveklabnik** [comment]
hmmm no errors in the console 
![2016-02-11-125828_622x569_scrot](https://cloud.githubusercontent.com/assets/27786/12985153/15ba59a0-d0bf-11e5-927a-8b59caf949a5.png)

**Comment by Manishearth** [comment]
@steveklabnik Clearly it's saying that we are so web that we are web itself :stuck_out_tongue_closed_eyes: 

That looks like a font issue.

**Comment by steveklabnik** [comment]
yeah i hard refreshed and it didn't change anything. and it says the font downloaded :/

**Comment by gnunicorn** [comment]
@steveklabnik: could you share your browser (version) – then I'll try to debug it? Also, running uBlock or any Adblocker blocking google fonts or something?

**Comment by gnunicorn** [comment]
(weird that 'web' shows up though. it is in an `<em>` but still the same font and everything...)

**Comment by steveklabnik** [comment]
Iceweasel 38.6.0, debian x86_64.

I don't use blockers. And it's odd because the main site works just fine.

**Comment by gnunicorn** [comment]
@steveklabnik I opened an separate issue about it: #5 – let's investigate it there.

**Comment by gnunicorn** [comment]
I've done some major updates today,
- added the about section
- added a news section (with news from the ecosystem)
- more docs
- rewrote the atom-feed
- toggle-able comments
- curators section
- I dropped the text-shadow, hopefully eradicating #5.

here is what it looks like today as of now – you can find it at http://rewrite.arewewebyet.org/

![screen shot 2016-02-12 at 19 44 12](https://cloud.githubusercontent.com/assets/40496/13007189/10d1ed74-d1c1-11e5-8c14-d67f0d3cc351.png)

---

The only things missing right now are: a proper review (especially of the text), a relaunch-blog-post and some more curators. Curators are basically the maintainers of the website: each one having full write access and we'd update the page regularly, merge PRs and discuss whether some news make us re-evaluate the state of web. Of course, curators are also very much invited to add their own comments and reviews, as well as spread the word about the website/relaunch.

**So, @Manishearth, @steveklabnik, @andradei, @frewsxcv : would you like to become curators**? I'd love to add all of you!

**Comment by frewsxcv** [comment]
Can I request https://github.com/briansmith/ring get added to the "crypto" section?

> ring is a crypto library in Rust based on BoringSSL's crypto primitive implementations.

As as a description should suffice

**Comment by frewsxcv** [comment]
Does the source for the rewrite live in this repo? Possibly https://github.com/bashyHQ/arewewebyet-rewrite ?

**Comment by gnunicorn** [comment]
> Does the source for the rewrite live in this repo? Possibly https://github.com/bashyHQ/arewewebyet-rewrite ?

@frewsxcv yes. github doesn't support hosting two websites (with different domains) from the same repo. So in order to have a preview for the rewrite while still running the old until then, I created a temp repo. Once all is done I will merge it back in here and delete the other repo. That's also why I kept all issues here.

**Comment by gnunicorn** [comment]
@steveklabnik , @Manishearth , @frewsxcv : is really no one interested in helping me curate (aka maintain) this?

**Comment by Manishearth** [comment]
Wait, I didn't say I was? Sorry!

I am certainly interested :smile:

**Comment by gnunicorn** [comment]
@Manishearth you are added to the curators and should have received an invite to join the team from github.

---

Meanwhile I merged the rewrite and published it along with the announcement blog post: http://www.arewewebyet.org/news/2016/02/16/we-are-back-baby/

Great work everyone!

**Comment by Manishearth** [comment]
Yep; got the invite!
